### PR TITLE
Add systemd unit drop-in example

### PR DESCRIPTION
**Pull request by eliasp**
- Describe how to modify existing services using a systemd unit drop-in
- Set syntax-highlighting for systemd unit examples to `INI`

### CODE REVIEW

**Comment by eliasp** [comment]
Sure - makes totally sense.

PR is updated.

File: doc/examples.md [hunk 1]
```diff
@@ -19,16 +21,42 @@ This config will write a single service unit (shown below) with the contents of
...
 [Service]
 Type=oneshot
 ExecStart=/usr/bin/echo Hello World
 
 [Install]
 WantedBy=multi-user.target
 ```
+### Modifying Services
+
+This config will add a [systemd unit drop-in](https://coreos.com/os/docs/latest/using-systemd-drop-in-units.html) to modify an existing service. It enables the `systemd-networkd` service and sets its environment variable `SYSTEMD_LOG_LEVEL` to `debug`.
```
Comment (crawford) [inline doc/examples.md]: `systemd-networkd` is typically already enabled. Go ahead and drop that part. Just the dropin should be fine.

File: doc/examples.md [hunk 2]
```diff
@@ -19,16 +21,42 @@ This config will write a single service unit (shown below) with the contents of
...
+
+```json
+{
+  "ignition": { "version": "2.0.0" },
+  "systemd": {
+    "units": [{
+      "name": "systemd-networkd.service",
+      "enable": true,
```
Comment (crawford) [inline doc/examples.md]: See above.

File: doc/examples.md [hunk 3]
```diff
@@ -19,16 +21,42 @@ This config will write a single service unit (shown below) with the contents of
...
+      "dropins": [{
+        "name": "environment.conf",
```
Comment (crawford) [inline doc/examples.md]: Can you name this "debug.conf"?

File: doc/examples.md [hunk 4]
```diff
@@ -19,16 +21,42 @@ This config will write a single service unit (shown below) with the contents of
...
+        "contents": "[Service]\nEnvironment=SYSTEMD_LOG_LEVEL=debug"
+      }]
+    }]
+  }
+}
+```
+
+#### systemd-networkd.service.d/environment.conf
```
Comment (crawford) [inline doc/examples.md]: See above.

File: doc/examples.md (hunk 1 above)
Comment (eliasp) [inline doc/examples.md]: Done

File: doc/examples.md (hunk 2 above)
Comment (eliasp) [inline doc/examples.md]: Done

File: doc/examples.md (hunk 3 above)
Comment (eliasp) [inline doc/examples.md]: Done

File: doc/examples.md (hunk 4 above)
Comment (eliasp) [inline doc/examples.md]: Done

**Comment by eliasp** [comment]
PR is updated according to your comments.

**Comment by crawford** [comment]
It doesn't look like the changes were added to the latest commit.

**Comment by eliasp** [comment]
Sorry about that - fixed now.

**Pull request merged**